fla/layers/mom_varlen.py

```
@@ -0,0 +1,749 @@
+        cu_q, cu_k, cu_v, cu_g, cu_beta, indices_q, cu_seqlen_all, max_seq_lens = _upad_input(q, k, v, g, beta, mask2, q_len)
+        cu_seqlen = cu_seqlen_all[0].to(torch.long).unique()
+        cu_q, cu_k, cu_v, cu_g, cu_beta = (x.unsqueeze(0).contiguous() for x in (cu_q, cu_k, cu_v, cu_g, cu_beta))
+
+        # dealing with padding
+        if attention_mask is not None:
+            beta = beta.mul(attention_mask[None, :, -beta.shape[-2]:, None])
+            g = g.mul(attention_mask[None, :, -g.shape[-2]:, None])
+
+        recurrent_state = last_state['recurrent_state'] if last_state is not None else [
+            None for _ in range(1 + self.shared_mem)]
+        offsets = kwargs.get('offsets', None)
```

Ruff (0.11.9)

Local variable `offsets` is assigned to but never used

Remove assignment to unused variable `offsets`

(F841)